### PR TITLE
[Method References 2] LPS-165169 Convert method reference to ServiceTrackerList in module portal-search | part 1

### PR DESCRIPTION
**Pull request by W2jx**
Detail: https://github.com/liferay-core-infra/liferay-portal/pull/2278

### CODE REVIEW

**Comment by zhangyan0701** [comment]
ci:test:search

**Comment by Kvale1733** [comment]
@W2jx No regressions found.

**Comment by W2jx** [comment]
Force-Push is just to rebase and resolve conflicts about files about 
```
modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/indexer/SearchPermissionFilterContributorsRegistryImpl.java
modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/searcher/SearchRequestContributorsRegistryImpl.java
```

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:search

**Comment by W2jx** [comment]
ci:test:sf

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by W2jx** [comment]
no conflicts(2022/12/5 13:20)

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/searcher/SearchRequestContributorsRegistryImpl.java [hunk 1]
```diff
@@ -99,7 +82,7 @@ private Collection<SearchRequestContributor> _include(
 		Collection<String> ids) {
 
 		if ((ids == null) || ids.isEmpty()) {
-			return new ArrayList<>(_searchRequestContributors);
+			return _serviceTrackerList.toList();
```
Comment (tinatian) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/searcher/SearchRequestContributorsRegistryImpl.java]: @W2jx Please remove service tracker list, collect the results by iterate the map's values.

**Comment by tinatian** [comment]
@W2jx Please see my comment and resend.

This is also a big pull, please separate and resend. When I review the second commit and I saw an issue, I will stop and NOT review your other changes, separating into small ones and you will get your code reviewed more.

Thanks.
Tina

**Comment by W2jx** [comment]
Hi @tinatian , thank you for reminding me. I will separate this pull.

**Pull request closed without merging**